Bins Role uses offline keys (#1)
secure-systems-lab/peps: pep-0458.txt

Figure 2: An overview of the role metadata available on PyPI.

The roles that change most frequently are *timestamp*, *snapshot* and roles

> delegated to by *bins*.  The *timestamp* and *snapshot*

Reviewer: ```suggestion
delegated to by *bins* (i.e., *bin-n*).  The *timestamp* and *snapshot*
```

metadata MUST be updated whenever *root*, *targets* or delegated metadata are
updated.  Observe, though, that *root* and *targets* metadata are much less
likely to be updated as often as delegated metadata.  Similarly, the *bins* role

> will only be updated when new bins are added.  Therefore, *timestamp*

Reviewer: ```suggestion
will only be updated when new bins are added.  Therefore, *timestamp*,
```

> and *snapshot* metadata will most likely be updated frequently (possibly every

Reviewer: ```suggestion
*snapshot*, and *bin-n* metadata will most likely be updated frequently (possibly every
```

protect projects from attackers who have compromised PyPI, since attackers can
manipulate TUF metadata using the keys stored online.

> This PEP proposes that the *bins* role's delegated roles sign for all

Reviewer: ```suggestion
This PEP proposes that the *bin-n* roles roles sign for all
```

> The *root* and *targets* role metadata SHOULD expire in one year, because these

Reviewer: ```suggestion
The metadata for the *root*, *targets*, and *bins* roles SHOULD each expire in one year, because these
```

two metadata files are expected to change very rarely.

> The *timestamp*, *snapshot*, and *bin-n* metadata SHOULD expire in one day

Reviewer: ```suggestion
The *timestamp*, *snapshot*, and *bin-n* metadata SHOULD each expire in one day
```
